### ISSUE TITLE
Move `app.request()`, `c.stream()` and `c.streamText()`

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#1843]
### What is the feature you are proposing?

For Hono, being small is as important as being fast.

In Cloudflare Workers, some benchmarks show that the smaller the size, the faster it is under certain conditions.

https://github.com/TigersWay/cloudflare-playground

This is an extreme example, but smaller is better.

Currently, the `hono/tiny` preset Hello World application is 13.04 KB after minify; the goal is to get it down to 10 KB in v4.

```txt
yusuke $ wrangler deploy --minify --dry-run src/tiny.ts
 ⛅️ wrangler 3.22.1
-------------------
--dry-run: exiting now.
Total Upload: 13.04 KiB / gzip: 5.05 KiB
```

I will do some minor refactoring, such as making common strings variables. However, to make it smaller, I would like to move `app.request()` to `hono/dev` and `c.stream()` and `c.streamText()` to `hono/streaming`.

### `app.request()`

There is not a lot of code in `app.request()`, but it does not have to be in `hono-base.ts`. This is because this function is mainly used for testing. So I propose to move it to `hono/dev`.

```ts
import { request } from 'hono/dev'

//... 

const res = await request(app, '/')
```

### `c.stream()` and `c.streamText()`

The `c.stream()` and `c.streamText()` functions are recently created and very attractive. However, they are used less often than `c.text()`, `c.json()`, and `c.html()`, and have slightly more code. If deleted, the size will be reduced by less than 1KB.

```txt
yusuke $ wrangler deploy --minify --dry-run src/tiny.ts
 ⛅️ wrangler 3.22.1
-------------------
--dry-run: exiting now.
Total Upload: 12.20 KiB / gzip: 4.76 KiB
```

Therefore, I propose to move them to `hono/streaming`.

```ts
import { stream } from 'hono/streaming'

app.get('/stream', (c) => {
  return stream(c, async (stream) => {
    // ...
  })
})
```

---

Since these are major breaking changes, I plan to deprecate the previous functions in the next minor update and remove them in v4. Breaking changes are painful for us, but one of the most important things to do is to keep it small, and if there are no major problems, we have to do it.

**Comment by yusukebe** [comment]
@sor4chi If you agree, can you create a PR to move `c.stream()` and `c.streamText()`?

@hagishi If you look at it and agree, can you create a PR to move the `app.request()`?

**Comment by watany-dev** [comment]
Regarding streaming helper, I think it is good to rename the current streaming/index.ts to streaming/sse.ts.

**Comment by yusukebe** [comment]
@watany-dev 

Yeah. You are right!

**Comment by yusukebe** [comment]
We may need to rethink the `app.request()` thing. Because more people use it than I could ever imagine or imagine.

Also, you can pass a `Request` object as the first argument to `app.request()`, but few people do that, and in that case, we could use `app.fetch()`, so we could change the API to accept only `string` and `URL`.

@hagishi What do you think?

**Comment by sor4chi** [comment]
I certainly agree with @yusukebe that there are few use cases for `c.stream` and `c.streamText`, since `c.body` is sufficient in most cases when using `ReadableStream`, as he says.
I’ll create PR to move to helper for v4.
Also, I’ll move the SSE helper that @watany created. thanks!

**Comment by hagishi** [comment]
@yusukebe 

Totally on board with that. 
Keeping the endpoint tests easy and clear is key, so just using a path string works great. 
No need to complicate things!

**Comment by yusukebe** [comment]
@hagishi 

Thanks! However, let's keep the current API for `app.request()` because some users pass the `Request` object to it, so we should avoid the breaking change.

**Comment by yusukebe** [comment]
Fixed by #1846. Thanks @sor4chi !
